src/main/java/org/javarosa/xform/parse/ExternalInstanceParser.java

```diff
@@ -36,6 +53,17 @@ public void addProcessor(Processor processor) {
         externalDataInstanceProcessors.add((ExternalDataInstanceProcessor) processor);
     }
 
+    /**
+     * Adds {@link FileInstanceParser} before others. The last added {@link FileInstanceParser} will be checked
+     * (via {@link FileInstanceParser#isSupported(String, String)}) first.
+     */
+    public void addFileInstanceParser(FileInstanceParser fileInstanceParser) {
+        fileInstanceParsers = Stream.concat(
+            Stream.of(fileInstanceParser),
+            fileInstanceParsers.stream()
+        ).collect(Collectors.toList());
```

Comment:
This got a little bit of an eyebrow raise from me. 😅 Is it taking a streaming approach too far? Do you want to avoid `fileInstanceParsers.add(0, fileInstanceParser)` to avoid direct mutation? Is it substantively different here? Is it to be consistent with other stream-based list processing?

I don't feel strongly but wanted to share my eyebrow raise.

Reply:
It's "completely forgetting about `add(0, fileInstanceParser)` and trying to append two lists". Remember we don't have to deal with any of this insanity in Collect any longer!

Reply:
Derp. Can't do that because it's an immutable list. Sigh.